pulsar-common/src/main/java/org/apache/pulsar/common/util/collections/LongPairRangeSet.java

```diff
@@ -125,6 +128,26 @@
      */
     Range<T> lastRange();
 
+    /**
+     * Retrieve internal-ranges {@link LongPairRangeSet} to byte[].
+     * 
+     * @param maxRanges
+     *            to retrieve limited ranges.
+     * @return
+     */
+    default Map<Long, long[]> toRanges(int maxRanges) {
+        return Collections.emptyMap();
```

Comment:
isn't it better to throw UnsupportedOperationException ?

also, if I retrieve only a subset of the data, how do I know that there would be more ranges ?